libraries/chain/transaction_context.cpp

```diff
@@ -128,6 +128,24 @@ namespace eosio { namespace chain {
             }
          }
       }
+
+      if( control.is_builtin_activated( builtin_protocol_feature_t::register_transaction_hook ) ) {
+         auto gprops = control.get_global_properties();
+         const uint32_t preexecution_hook_type = 0;
+
+         auto it = find_if(std::begin(gprops.transaction_hooks), std::end(gprops.transaction_hooks),
+                           [&](transaction_hook &t_hook) {
+                              return t_hook.type == preexecution_hook_type;
+                           }
+         );
+
+         if (it != std::end(gprops.transaction_hooks)) {
+            chain::action preexecution_hook_action{{}, it->contract, it->action, {}};
+            schedule_action(preexecution_hook_action, preexecution_hook_action.account, false, 0, 0);
+            execute_action( 1, 0 );
```

Comment:
Not sure you want to do this. It will generate traces for the preexecution_hook_action. Also not sure the ordinals are correct. Need to determine what if anything should be logged to deep mind logger for this.

Reply:
I modified my code so that it doesn't add the action trace of the preexecution hook, but i am not 100% sure understanding why this is not required, if you can elaborate on why is this the case,  that would be helpful.

Also I removed the part of the logging to deep mind.

Any other feedback about how to better do this task is welcome.

Reply:
I don't know what the requirements are for an action trace for this preexecution_hook. Need to ask @b1bart if an action trace is desired. And if desired if it should be logged to deep mind. I would suspect that this extra action trace would be confusing to consumers of transaction traces.

Reply:
@heifner, after consulting @b1bart, the trace should be included for preexecution_hook action and send to deep mind.